internal/endpoints/awsrulesfn/partition.go

```diff
@@ -0,0 +1,68 @@
+package awsrulesfn
+
+import "regexp"
+
+// Partition provides the metadata describing an AWS partition.
+type Partition struct {
+	ID            string                     `json:"id"`
+	Regions       map[string]PartitionConfig `json:"regions"`
+	RegionRegex   string                     `json:"regionRegex"`
+	DefaultConfig PartitionConfig            `json:"outputs"`
+}
+
+// PartitionConfig provides the endpoint metadata for an AWS region or partition.
+type PartitionConfig struct {
+	Name               *string `json:"name"`
+	DNSSuffix          *string `json:"dnsSuffix"`
+	DualStackDNSSuffix *string `json:"dualStackDnsSuffix"`
+	SupportsFIPS       *bool   `json:"supportsFIPS"`
+	SupportsDualStack  *bool   `json:"supportsDualStack"`
+}
+
+const defaultPartition = "aws"
+
+// GetPartition returns an AWS [Partition] for the region provided. If the
+// partition cannot be determined nil will be returned.
+func GetPartition(partitions []Partition, region string) *PartitionConfig {
+	for _, partition := range partitions {
+		if v, ok := partition.Regions[region]; ok {
+			v.mergeWith(partition.DefaultConfig)
```

Comment:
**Question**: I think this looks correct but just to clarify this is creating a copy of the struct from the partitions array and merging into the copy and not the original one right? (i.e. we aren't accidentally mutating the partitions in the partition array)

Reply:
gotcha, yah its correct. i even tested it to confirm its a copy

Reply:
since the performance here seems negligible, i altered the function to make the copying more explicit (it was also bothering me that when i quickly glance at it, it makes me do a double take to see if its actually copying)